data.py

```diff
@@ -35,119 +36,185 @@ def set_context(resource_item):
     out_dir.mkdir()
 
+    # Specify locations of JSON files
+    datasets_source_path = Path(metadata_stub_content_path / "datasets.json")
+    editions_source_path = Path(metadata_stub_content_path / "editions")
+    versions_source_path = Path(metadata_stub_content_path / "editions" / "versions")
+    topics_source_path = Path(metadata_stub_content_path / "topics.json")
+    publishers_source_path = Path(metadata_stub_content_path / "publishers.json")
+
     # ------------------
     # Datasets resources
     # ------------------
 
-    # Load from disk
-    datasets_source_path = Path(subbed_metadata_store_content_path / "datasets.json")
+    # Load json file from disk
     with open(datasets_source_path) as f:
         datasets_source_dict = json.load(f)
-        # Validate then add to graph
-        schemas.Datasets(**datasets_source_dict)
-        g += Graph().parse(
-            data=json.dumps(set_context(datasets_source_dict)),
-            format="json-ld",
-        )
-    assert len(g) > graph_length
-    graph_length = len(g)
+
+    # Validate data and add to graph
+    validate_and_parse_json(g, schemas.Datasets, datasets_source_dict, "datasets")
+
+    # Extract edition URLs from datasets.json for validation later
+    dataset_editions_urls = [
+        dataset["editions_url"] for dataset in datasets_source_dict["datasets"]
+    ]
+
+    # Extract publishers from datasets.json for validation later
+    dataset_publishers = {
+        dataset["publisher"] for dataset in datasets_source_dict["datasets"]
+    }
 
     # ------------------
     # Editions resources
     # ------------------
 
-    # Load from disk
-    editions_source_path = Path(subbed_metadata_store_content_path / "editions")
-    # Validate then add to graph
-    schema = schemas.Editions
-    process_json_files(g, editions_source_path, schema)
+    # Load json files from disk
+    editions = process_json_files(editions_source_path)
+
+    # Empty dict to hold version data from editions for validation later
+    versions_in_edition = dict()
+    # Empty set to hold topic data from editions for validation later
+    topics_in_editions = set()
+
+    # Validate data and add to graph
+    for edition in editions:
+        assert (
+            edition["@id"] in dataset_editions_urls
+        ), f"Editions URL {edition['@id']} not found in {dataset_editions_urls}"
+        for edn in edition["editions"]:
+            summarised_editions = [
+                {
+                    "@id": edn["@id"],
+                    "issued": edn["issued"],
+                    "modified": edn["modified"],
+                }
+            ]
+            versions_in_edition[edn["versions_url"]] = edn["versions"]
+            assert (
+                edn["publisher"] in dataset_publishers
+            ), f"{edn['publisher']} not in publisher list {dataset_publishers}"
+            topics_in_editions.update(edn["topics"])
+        for dataset in datasets_source_dict["datasets"]:
```

Comment:
The nested for loops here might make the function drag on a bit, I personally don't mind but maybe it could benefit from some abstraction/refactoring at this part? But it might also be a bit redundant to just split things up. You know best with this populate function so let me know what you think.